lib/kaguya/core.ex

```diff
@@ -128,6 +133,7 @@ defmodule Kaguya.Core do
     cancel_server_timer(state)
     socket = reconnect()
     send(self(), :init)
+    send(self(), :rejoin_chan)
```

Comment:
Ah good point, I forgot that it spawned a Task to send the values. If might be worth making that bit of code synchronous to simplify this.